### PR TITLE
Allow using latest Breathe with Sphinx 8.x

### LINKED ISSUE #63: No breathe for sphinx 8.x
### Comment:

Is it possible to bump the pin on Sphinx?

### PR DESCRIPTION
**Pull request by vyasr**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closes #63 

[Breathe is testing against Sphinx 8 in their CI](https://github.com/breathe-doc/breathe/blob/main/.github/workflows/unit_tests.yml#L45), so this update should be safe.

### CODE REVIEW

**Comment by vyasr** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml
Comment (bdice) [inline recipe/meta.yaml:1]: Could we update the homepage and dev URL at the bottom of this file? I think the canonical URL is now https://github.com/breathe-doc/breathe.

Comment (vyasr) [reply recipe/meta.yaml:1]: Done.

**Comment by vyasr** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
Hmm sorry i merged this: https://github.com/conda-forge/breathe-feedstock/pull/65

**Comment by hmaarrfk** [comment]
often i rebase deleting any commit that starts with MNT.

If you want you can recreate the PR or I can.

**Comment by vyasr** [comment]
Not a problem I'll rebase.

**Comment by vyasr** [comment]
@conda-forge-admin please rerender

**Comment by vyasr** [comment]
@hmaarrfk I think we should be good now.

**Comment by vyasr** [comment]
Thanks!

**Comment by bdice** [comment]
@vyasr @hmaarrfk I think we missed bumping the build number here after rebasing. We will need to bump that to get new packages.

**Comment by vyasr** [comment]
Git was too smart for me and cleanly merged that change since it was identical on both branches. I opened #66

**Pull request merged**